### PR TITLE
[Form] Add choice_translation_parameters option

### PR DESCRIPTION
**Pull request by VincentLanglet**
Cherry pick of https://github.com/symfony/symfony-docs/pull/13677

The feature is merged https://github.com/symfony/symfony/pull/38469

### CODE REVIEW

File: reference/forms/types/choice.rst [hunk 1]
```diff
@@ -230,7 +231,13 @@ correct types will be assigned to the model.
 
 .. include:: /reference/forms/types/options/choice_name.rst.inc
 
+<<<<<<< HEAD
```
Comment (OskarStark) [inline reference/forms/types/choice.rst]: Can you please solve the merge conflict? Thanks

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: reference/forms/types/choice.rst [hunk 2]
```diff
@@ -230,7 +231,13 @@ correct types will be assigned to the model.
...
 .. include:: /reference/forms/types/options/choice_translation_domain_enabled.rst.inc
```
Comment (ricardodevries) [inline reference/forms/types/choice.rst:235]: Please add a blank line before this include.

File: reference/forms/types/choice.rst [hunk 3]
```diff
@@ -230,7 +231,13 @@ correct types will be assigned to the model.
...
+=======
+.. include:: /reference/forms/types/options/choice_translation_parameters.rst.inc
```
Comment (ricardodevries) [inline reference/forms/types/choice.rst:237]: Please add a blank line before this include.

**Review by ricardodevries — CHANGES_REQUESTED** [review]

File: reference/forms/types/choice.rst (hunk 1 above)
Comment (VincentLanglet) [reply reference/forms/types/choice.rst]: Solved

**Comment by VincentLanglet** [comment]
Friendly ping @OskarStark is it now ok for you ?

**Review by OskarStark — APPROVED** [review]

File: reference/forms/types/choice.rst [hunk 4]
```diff
@@ -18,6 +18,7 @@ To use this field, you must specify *either* ``choices`` or ``choice_loader`` op
 |                           | - `choice_label`_                                                    |
 |                           | - `choice_loader`_                                                   |
 |                           | - `choice_name`_                                                     |
+|                           | - `choice_translation_parameters`_                                   |
```
Comment (xabbuh) [inline reference/forms/types/choice.rst]: should be moved below `choice_translation_domain`

File: reference/forms/types/choice.rst [hunk 5]
```diff
@@ -230,6 +231,8 @@ correct types will be assigned to the model.
 
 .. include:: /reference/forms/types/options/choice_name.rst.inc
 
+.. include:: /reference/forms/types/options/choice_translation_parameters.rst.inc
```
Comment (xabbuh) [inline reference/forms/types/choice.rst]: same here

File: reference/forms/types/options/choice_translation_parameters.rst.inc [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            'form.order.yes' => true,
+            'form.order.no' => false,
+        ],
+        'choice_translation_parameters' => [
+            'form.order.yes' => ['%company%' => 'ACME Inc.'],
+            'form.order.no' => [],
+        ],
+    ]);
+
+The ``choice_translation_parameters`` option of children fields is merged with
```
Comment (xabbuh) [inline reference/forms/types/options/choice_translation_parameters.rst.inc]: "The translation parameters of child fields are [...]"?

File: reference/forms/types/options/choice_translation_parameters.rst.inc [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+The choice values are translated before displaying it, so it can contain
+:ref:`translation placeholders <component-translation-placeholders>`.
+This option defines the values used to replace those placeholders. This can be
+an associative array where the keys match the choice keys and the values
+are the attributes for each choice, a callable or a property path
+(just like `choice_label`_).
+
+Given this translation message:
+
+.. code-block:: yaml
```
Comment (xabbuh) [inline reference/forms/types/options/choice_translation_parameters.rst.inc]: What about adding the other translation formats as well?

**Comment by OskarStark** [comment]
@VincentLanglet could you please apply the comments made by @xabbuh ? Thanks!

File: reference/forms/types/options/choice_translation_parameters.rst.inc (hunk 7 above)
Comment (VincentLanglet) [reply reference/forms/types/options/choice_translation_parameters.rst.inc]: Done

File: reference/forms/types/options/choice_translation_parameters.rst.inc (hunk 6 above)
Comment (VincentLanglet) [reply reference/forms/types/options/choice_translation_parameters.rst.inc]: Done

File: reference/forms/types/choice.rst (hunk 5 above)
Comment (VincentLanglet) [reply reference/forms/types/choice.rst]: Done

File: reference/forms/types/choice.rst (hunk 4 above)
Comment (VincentLanglet) [reply reference/forms/types/choice.rst]: Done

**Comment by VincentLanglet** [comment]
> @VincentLanglet could you please apply the comments made by @xabbuh ? Thanks!

It should be ok now ; but I got an unrelated error @OskarStark

**Review by OskarStark — APPROVED** [review]

**Comment by xabbuh** [comment]
@VincentLanglet Can you rebase here please?

**Comment by VincentLanglet** [comment]
Sure, it's done @xabbuh

**Review by xabbuh — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Vincent.

**Pull request merged**